### PR TITLE
doc: how to configure API/APP key with OLM

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -50,6 +50,48 @@ spec:
...
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: my-datadog-operator
+  namespace: operators
+spec:
+  channel: alpha
+  name: datadog-operator
+  source: operatorhubio-catalog
+  sourceNamespace: olm
```
Comment (clamoriniere) [inline docs/installation.md:78]:
```suggestion
  # ...
```

File: docs/installation.md [hunk 2]
```diff
@@ -50,6 +50,48 @@ spec:
...
+    env:
+      - name: DD_API_KEY
+        valueFrom:
+          secretKeyRef: 
+             key: api-key
+             name: datadog-secret
+      - name: DD_APP_KEY
+        valueFrom:
+          secretKeyRef: 
+            key: api-ky
```
Comment (davidor) [inline docs/installation.md:89]: `app-key`

File: docs/installation.md [hunk 3]
```diff
@@ -50,6 +50,48 @@ spec:
...
+            name: datadog-secret
+```
+
+**Note**: several config overrides can be added on the `spec.config` section; example: `env` + `resources`
+
 ## Deploy the Datadog Agents with the operator
```
Comment (alai97) [inline docs/installation.md:95]:
```suggestion
## Deploy the Datadog Agents with the Operator
```

File: docs/installation.md (hunk 2 above)
Comment (alai97) [reply docs/installation.md:89]: Good catch!

```suggestion
            key: app-key
```

File: docs/installation.md [hunk 4]
```diff
@@ -50,6 +50,48 @@ spec:
         cpu: "500m"
 ```
 
+### Add Datadog credential to the operator
+
+For some usecase (example: `DatadogMonitors` support), the operator requires to have access to a Datadog API and APP keys.
+
+Create a secret that contains this two keys. In the following example the secret keys are `api-key` and `app-key`.
```
Comment (alai97) [inline docs/installation.md:57]:
```suggestion
The Datadog Operator requires access to your API and application keys to add a `DatadogMonitor`. 

1. Create a secret that contains both keys. In the example below, the secret keys are `api-key` and `app-key`.
```

File: docs/installation.md [hunk 5]
```diff
@@ -50,6 +50,48 @@ spec:
...
+
+```
+export DD_API_KEY=<replace-by-your-api-key>
+export DD_APP_KEY=<replace-by-your-app-key>
+
+kubectl create secret generic datadog-secret --from-literal api-key=$DD_API_KEY --from-literal app-key=$DD_APP_KEY
+```
+
+Add the references to the secret in the Datadog-Operator Subscription resource instance.
```
Comment (alai97) [inline docs/installation.md:66]:
```suggestion
2. Add references to the secret in the Datadog-Operator Subscription resource instance. 
```

File: docs/installation.md [hunk 6]
```diff
@@ -50,6 +50,48 @@ spec:
...
+             key: api-key
+             name: datadog-secret
+      - name: DD_APP_KEY
+        valueFrom:
+          secretKeyRef: 
+            key: api-ky
+            name: datadog-secret
+```
+
+**Note**: several config overrides can be added on the `spec.config` section; example: `env` + `resources`
```
Comment (alai97) [inline docs/installation.md:93]:
```suggestion
**Note**: You can add configuration overrides in the `spec.config` section. For example, override `env` and `resources`.
```

**Review by alai97 — CHANGES_REQUESTED** [review]

**Comment by clamoriniere** [comment]
hi @davidor and @alai97,
I have updated the PR according to your suggestions. It is ready for a second review :)

**Review by alai97 — APPROVED** [review]

**Pull request merged**